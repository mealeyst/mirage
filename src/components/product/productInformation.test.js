import React from 'react'
import 'jest-styled-components'
import { ProductInformation, H1, H2, H3, P } from 'SRC'

const { mountWithTheme } = global

const defaultProps = {
  subheader: "The Trend: Sparkly Rainbow",
  header: "Weekends Are For Fun Outfit",
  price: "$55",
  description: "Cheesecake bocconcini red leicester. Melted cheese when the cheese comes out everybody's happy."
}

describe('() ProductInformation', () => {
  const createProductInformation = (inProps) => {
    const props = {
      ...defaultProps,
      ...inProps
    }
    return mountWithTheme(<ProductInformation {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createProductInformation())
    .toMatchSnapshot()
  })

  test('header should always render', () => {
    expect(createProductInformation().find(H1).text()).toMatch(defaultProps.header)
  })

  test('price should always render', () => {
    expect(createProductInformation().find(H2).text()).toMatch(defaultProps.price)
  })

  describe('subheader', () => {
    test("shouldn't render if missing prop", () => {
      const props = {
        subheader: undefined
      }
      expect(createProductInformation(props).find(H3).length).toEqual(0)
    })

    test('should render if provided', () => {
      expect(createProductInformation().find(H3).text()).toMatch(defaultProps.subheader)
    })
  })

  describe('description', () => {
    test("shouldn't render if missing prop", () => {
      const props = {
        description: undefined
      }
      expect(createProductInformation(props).find(P).length).toEqual(0)
    })

    test('should render if provided', () => {
      expect(createProductInformation().find(P).text()).toMatch(defaultProps.description)
    })
  })
})