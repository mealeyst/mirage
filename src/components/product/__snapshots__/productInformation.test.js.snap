// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`() ProductInformation matching the snapshot 1`] = `
.c4 {
  -webkit-letter-spacing: .1rem;
  -moz-letter-spacing: .1rem;
  -ms-letter-spacing: .1rem;
  letter-spacing: .1rem;
  text-transform: uppercase;
  color: #00003C;
  font-family: "din-cond","din-condensed-web",Arial,sans-serif;
  font-size: 3.2rem;
  font-weight: 500;
  line-height: 1.3;
  margin: 5.5rem 0;
}

.c6 {
  -webkit-letter-spacing: .05rem;
  -moz-letter-spacing: .05rem;
  -ms-letter-spacing: .05rem;
  letter-spacing: .05rem;
  text-transform: uppercase;
  color: #00003C;
  font-family: "din-cond","din-condensed-web",Arial,sans-serif;
  font-size: 3.2rem;
  font-weight: 500;
  line-height: 1.0625;
  margin: 2.6rem 0;
}

.c2 {
  -webkit-letter-spacing: .05rem;
  -moz-letter-spacing: .05rem;
  -ms-letter-spacing: .05rem;
  letter-spacing: .05rem;
  text-transform: uppercase;
  color: #00003C;
  font-family: "din-cond","din-condensed-web",Arial,sans-serif;
  font-size: 2.4rem;
  font-weight: 500;
  line-height: 1.0833333333333333;
  margin: 2.4rem 0;
}

.c9 {
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.6rem;
  line-height: normal;
  margin: 0;
  color: #00003C;
}

.c8 {
  height: .1rem;
  width: 100%;
  border: none;
  margin: 1rem 0;
}

.c7 {
  background-color: #D5D5D5;
}

.c0 .c3 {
  margin-top: 0;
  margin-bottom: 0.5rem;
}

.c0 .c5,
.c0 .c1 {
  margin-top: 2rem;
  margin-bottom: 1rem;
}

@media (min-width:768px) {
  .c4 {
    font-size: 4.2rem;
  }
}

@media (min-width:768px) {
  .c4 {
    line-height: 1.0476190476190477;
  }
}

<Styled(Component)
  description="Cheesecake bocconcini red leicester. Melted cheese when the cheese comes out everybody's happy."
  header="Weekends Are For Fun Outfit"
  price="$55"
  subheader="The Trend: Sparkly Rainbow"
>
  <Component
    className="c0"
    description="Cheesecake bocconcini red leicester. Melted cheese when the cheese comes out everybody's happy."
    header="Weekends Are For Fun Outfit"
    price="$55"
    subheader="The Trend: Sparkly Rainbow"
  >
    <div
      className="c0"
    >
      <styled.h3
        fontSize="2.4rem"
        margin="2.4rem 0"
      >
        <h3
          className="c1 c2"
          fontSize="2.4rem"
        >
          The Trend: Sparkly Rainbow
        </h3>
      </styled.h3>
      <styled.h1
        fontSizes={
          Object {
            "desktop": "4.2rem",
            "mobile": "3.2rem",
          }
        }
        lineHeights={
          Object {
            "desktop": 1.0476190476190477,
            "mobile": 1.3,
          }
        }
        margin="5.5rem 0"
      >
        <h1
          className="c3 c4"
        >
          Weekends Are For Fun Outfit
        </h1>
      </styled.h1>
      <styled.h2
        fontSize="3.2rem"
        margin="2.6rem 0"
      >
        <h2
          className="c5 c6"
          fontSize="3.2rem"
        >
          $55
        </h2>
      </styled.h2>
      <Styled(Styled(UnstyledHr))>
        <Styled(UnstyledHr)
          className="c7"
          width="100%"
        >
          <UnstyledHr
            className="c7 c8"
            width="100%"
          >
            <hr
              className="c7 c8"
            />
          </UnstyledHr>
        </Styled(UnstyledHr)>
      </Styled(Styled(UnstyledHr))>
      <styled.p
        color="#00003C"
        fontSize="1.6rem"
      >
        <p
          className="c9"
          color="#00003C"
          fontSize="1.6rem"
        >
          Cheesecake bocconcini red leicester. Melted cheese when the cheese comes out everybody's happy.
        </p>
      </styled.p>
    </div>
  </Component>
</Styled(Component)>
`;
